client/src/components/WorkflowInvocationState/WorkflowInvocationShare.vue

```diff
@@ -0,0 +1,142 @@
+<script setup lang="ts">
+import { faShareAlt } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BAlert, BButton, BModal } from "bootstrap-vue";
+import { computed, ref } from "vue";
+
+import { GalaxyApi } from "@/api";
+import { hasImportable } from "@/api/histories";
+import { getFullAppUrl } from "@/app/utils";
+import { Toast } from "@/composables/toast";
+import { useWorkflowInstance } from "@/composables/useWorkflowInstance";
+import { useHistoryStore } from "@/stores/historyStore";
+import { copy } from "@/utils/clipboard";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import LoadingSpan from "../LoadingSpan.vue";
+
+const CLIPBOARD_MSG = "The link to the invocation has been copied to your clipboard.";
+
+const props = defineProps<{
+    invocationId: string;
+    workflowId: string;
+    historyId: string;
+}>();
+
+const modalToggle = ref(false);
+
+// Workflow and History refs
+const { workflow, loading, error, owned } = useWorkflowInstance(props.workflowId);
+const historyStore = useHistoryStore();
+const history = computed(() => historyStore.getHistoryById(props.historyId));
+
+/** The link to the invocation. */
+const invocationLink = computed(() => getFullAppUrl(`workflows/invocations/${props.invocationId}`));
+
+/** If the history is already shareable.
+ * (i.e. It has the `importable` property, and it is set to `true`)
+ */
+const historyAlreadyShareable = computed(() => {
+    // Note: We will mostly have a summarized history object without the importable property
+    //       and in that case we still attempt to make it shareable.
+    return !!history.value && hasImportable(history.value) && history.value.importable;
```

Comment:
```suggestion
    return Boolean(history.value) && hasImportable(history.value) && history.value.importable;
```

Very minor comment, but I've seen this a couple of times and wanted to raise awareness about the cool style guide that we have :wink: 

https://github.com/galaxyproject/galaxy/blob/dev/client/docs/styleguide.md#casting

Reply:
That's a good point, thank you, I will make sure to do these right

Reply:
However, this causes this:
![image](https://github.com/user-attachments/assets/1355cf98-b787-44f3-87db-cc06fa894420)

Does this work better instead then?:
```suggestion
    return history.value && hasImportable(history.value) && history.value.importable;
```

Reply:
Either that or you can improve `hasImportable` to handle null values too: `hasImportable(entry?: AnyHistory | null)`, whatever feels better to you :)

Reply:
https://github.com/galaxyproject/galaxy/pull/19959/commits/3534cae7e1b6b4ef19cb70bbe2beb08867a0c672

Please let me know if the changes here seem correct? 😃

Reply:
Looks good! thank you!